test/kubernetes/integration_test.go

```diff
@@ -108,18 +108,236 @@ func TestPod_Single_Volume(t *testing.T) {
 		},
 	}
 
-	fmt.Println("Creating pvc")
+	t.Log("Creating pvc")
 	_, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(pvc)
 	if err != nil {
 		t.Fatal(err)
 	}
 
-	fmt.Println("Waiting pod to be running ...")
+	t.Logf("Waiting pod %q to be running ...\n", pod.Name)
 	if err := waitForPod(client, pod.Name); err != nil {
 		t.Error(err)
 	}
 
-	fmt.Println("Finished!")
+	t.Log("Finished!")
+}
+
+func TestDeployment_Single_Volume(t *testing.T) {
+	volumeName := "my-do-volume"
+	claimName := "csi-deployment-pvc"
+	appName := "my-csi-app"
+
+	replicaCount := new(int32)
+	*replicaCount = 1
+
+	dep := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: appName,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: replicaCount,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"app": appName,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": appName,
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "my-busybox",
+							Image: "busybox",
+							VolumeMounts: []v1.VolumeMount{
+								{
+									MountPath: "/data",
+									Name:      volumeName,
+								},
+							},
+							Command: []string{
+								"sleep",
+								"1000000",
+							},
+						},
+					},
+					Volumes: []v1.Volume{
+						{
+							Name: volumeName,
+							VolumeSource: v1.VolumeSource{
+								PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+									ClaimName: claimName,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	t.Log("Creating deployment")
+	_, err := client.AppsV1().Deployments(namespace).Create(dep)
```

Comment:
Any specific reason we are creating a deployment here for the test vs just a pod?

Reply:
Pod is another test (See above in the file). This is Deployment + PVC, we already test for Pod + PVC.

Reply:
@nanzhong if you're asking for the difference, pod doesn't get rescheduled, deployment will reschedule pod if pod get killed. We're going to add more test cases, such as removing a node and observe how a pod is getting scheduled. This will not be possible for pod based tests.